wp-content/themes/wp-starter/plopfile.js

```diff
@@ -0,0 +1,118 @@
+module.exports = function (plop) {
+
+	// Block icons from https://developer.wordpress.org/resource/dashicons/
+	const WPIcons = [
+		"default",
+		"align-left",
+		"cover-image",
+		"embed-video",
+		"format-gallery",
+		"layout",
+		"schedule",
+	]
+
+	// Block generator
+	plop.setGenerator('block', {
+		description: 'Set up a new block',
+		prompts: [
+			{
+				type: 'input',
+				name: 'name',
+				message: 'What is the block name?',
+				validate: function (value) {
+					if (/.+/.test(value)) {
+					  return true;
+					}
+					return "Name is required";
+				  },
+			},
+			{
+				type: 'input',
+				name: 'themeslug',
+				message: 'What is the slug for your theme?',
+				validate: function (value) {
+					if (/.+/.test(value)) {
+					  return true;
+					}
+					return "Theme slug is required";
+				  },
+			},
+			{
+				type: 'list',
+				name: 'icon',
+				message: 'Pick a WordPress icon for the block',
+				choices: WPIcons,
+			},
+			{
+				type: 'confirm',
+				name: 'styles',
+				message: 'Do you need block styles?'
+			},
+			{
+				type: 'confirm',
+				name: 'variations',
+				message: 'Do you need block variations?'
+			},
+		],
+		actions: function(data) {
+			var actions = [
+				{
+					type: 'addMany',
+					destination: 'blocks/{{dashCase name}}/',
+					base: 'plop-templates',
+					templateFiles: 'plop-templates/*.hbs',
+					abortOnFail: true,
+				},
+				{
+					type: 'addMany',
+					destination: 'blocks/{{dashCase name}}/',
+					base: 'plop-templates',
+					templateFiles: 'plop-templates/patterns/*.hbs',
+					abortOnFail: true,
+				},
+				{
+					type: "append",
+					path: "src/styles/main.css",
```

Comment:
Not sure if anything can be done about this, but when I deleted my `demo-block` directory, the import still remains here. Is there not a CSS library we can use that allows us to import entire folders dynamically so we don't have to specify every block CSS file, but rather use something like `blocks/**/{view,style}.css`?

Reply:
I looked into several way to do exactly that, but kept getting blocked. We would have to do something in Vite.
The other idea I had was to create a delete script that you could run and it would remove the files and the link in the css.